transport/conn.go

```diff
@@ -81,9 +96,12 @@ func (c *connWriter) loop() {
 				return
 			}
 			c.stats.inQueue.Dec()
-			if err := c.send(r); err != nil {
-				log.Printf("%s fatal send error, closing connection due to %s", c.connString(), err)
+			if err := c.send(ctx, r); err != nil {
 				r.ResponseHandler <- response{Err: fmt.Errorf("%s send: %w", c.connString(), err)}
+				if _, ok := err.(*skippedError); ok {
+					continue
```

Comment:
Wouldn't it casue streamIDs leak?

Reply:
ConnWriter should be able to free streamID. Some kind of a callback with pointer to connReader seems to be convenient solution, but I'm not sure if it is the best.

Reply:
Good point, this is how i'd like to resolve it, as in sendRequest we can't be sure we encounter a skippedError before looking at ctx.Done().